File: src/test/java/no/nav/tjenestepensjon/simulering/config/ClaimSetVerifierConfigTest.java
package no.nav.tjenestepensjon.simulering.config;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertThrows;

import static no.nav.tjenestepensjon.simulering.config.ClaimSetVerifierConfig.IssuerClaimVerifier;

import java.util.Map;
import java.util.Set;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;

class ClaimSetVerifierConfigTest {

    private static Map<String, Object> claims;

    @BeforeAll
    static void beforeAll() {
        claims = Map.of(
                "iss", "http://localhost.com:8080"
        );
    }

    @Test
    void shouldValidateIssuer() {
        IssuerClaimVerifier invalid = new IssuerClaimVerifier(Set.of("http://bogusissuer.com"));
        assertThrows(InvalidTokenException.class, () -> invalid.verify(claims));

        IssuerClaimVerifier valid = new IssuerClaimVerifier(Set.of("http://localhost.com:8080"));
        assertDoesNotThrow(() -> valid.verify(claims));
    }
}

File: src/test/kotlin/no/nav/tjenestepensjon/simulering/config/ClaimSetVerifierConfigTest.kt
package no.nav.tjenestepensjon.simulering.config

import no.nav.tjenestepensjon.simulering.config.ClaimSetVerifierConfig.IssuerClaimVerifier
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertDoesNotThrow
import org.junit.jupiter.api.assertThrows
import org.springframework.security.oauth2.common.exceptions.InvalidTokenException

internal class ClaimSetVerifierConfigTest {

    private val validIssuer = "http://localhost.com:8080"
    private val claims: Map<String, Any> = mapOf("iss" to validIssuer)

    @Test
    fun `Should validate issuer`() {
        val invalid = IssuerClaimVerifier(setOf("http://bogusissuer.com"))
        assertThrows<InvalidTokenException> { invalid.verify(claims) }
        val valid = IssuerClaimVerifier(setOf(validIssuer))
        assertDoesNotThrow { valid.verify(claims) }
    }
}